packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,335 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	tokens,
+} from "@fluentui/react-components";
+import {
+	DataEdit,
+	EditType,
+	FluidObjectValueNode,
+	Primitive,
+} from "@fluid-experimental/devtools-core";
+
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Input to EditableComponents
+ */
+interface EditableComponentProps {
+	node: FluidObjectValueNode;
+	setEditing: React.Dispatch<React.SetStateAction<boolean>>;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Render data that is Editable with its corresponding UI
+ *
+ * @remarks {@link MessageRelayContext} must be set in order to use this component.
+ */
+export function EditableView(props: EditableViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+
+	/**
+	 * State to store whether the user is currently editing data
+	 */
+	const [isEditing, setIsEditing] = React.useState<boolean>(false);
+
+	/**
+	 * State to store the current type, or the type of the intended edit
+	 */
+	const [type, setType] = React.useState<string>(typeof node.value);
```

Comment:
Nit: since this describes the type of the edit being made, maybe `editType`? What do you think?

Reply:
My thinking is that it might eventually describe the current type, and only the edit type if you start editing/ change the type. Fairly minor, I can change it. I think it was giving wierd clashes anyways due to the name "type" I think.